internal/pkg/deploy/cloudformation/env.go

```diff
@@ -153,30 +153,59 @@ func (cf CloudFormation) upgradeEnvironment(in *deploy.CreateEnvironmentInput, t
 	}
 
 	for {
-		// Set the parameters of the stack.
 		descr, err := cf.cfnClient.Describe(s.Name)
 		if err != nil {
 			return fmt.Errorf("describe stack %s: %w", s.Name, err)
 		}
+
+		if cloudformation.StackStatus(aws.StringValue(descr.StackStatus)).InProgress() {
+			// There is already an update happening to the environment stack.
+			// Best-effort try to wait for the existing update to be over before retrying.
+			_ = cf.cfnClient.WaitForUpdate(s.Name)
```

Comment:
Why don't we handle this error?

Reply:
it's best-effort, if for some reason the wait does not succeed it's no big deal, we will `continue` loop again, do a `Describe` find out that the stack is updating and try doing a `WaitForUpdate` again.